statsmodels/genmod/generalized_linear_model.py

```diff
@@ -41,7 +41,13 @@
 
 
 def _check_convergence(criterion, iteration, tol):
-    return not (np.fabs(criterion[iteration] - criterion[iteration-1]) > tol)
+    if iteration == 0:
+        return False
```

Comment:
I think you are right in general. We need one extra iteration to check convergence even with gaussian and OLS, because we don't know yet that we already convert. (I had forgotten about this.)

However, following your off-by-1 argument, I think the initial np.inf is redundant, since we initialized with the starting estimate. If the starting values (e.g. user provided start_params) are already optimal, then we only need the iteration=0 round to verify that we are converged.